src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -442,6 +452,9 @@ async Task SetArrayItemNode(object? d)
     protected virtual IFieldResolver SelectResolver(ExecutionNode node, ExecutionContext context)
         => node.FieldDefinition.Resolver ?? (node.Parent is RootExecutionNode && context.Operation.Operation == OperationType.Subscription ? SourceFieldResolver.Instance : NameFieldResolver.Instance);
 
+    [ThreadStatic]
+    private static ReadonlyResolveFieldContext? _readonlyResolveFieldContext;
```

Comment:
Don't you think `ObjectPool` will give better results? With `ThreadStatic`, you can reuse no more than number-of-threads objects, while there might be order of magnitude more concurrent tasks that need the context objects. It also can happen that one instance was promoted to a higher Gen but will be thrown out on the next cycle because another instance, probably from a younger generation, took its place. Then this young instance will be promoted to higher gen and thrown out again.

Reply:
No I don’t think so. At least, not here. The object in question only contains a two (?) values and is probably the minimum size of any allocation.  So it’s the same as any lambda or Task instance. ObjectPool is costly and only recommended when the cost to allocate is high.  And if the user needs to run asynchronous code to execute this field, one more allocation won’t perceptibly change the speed. ThreadStatic is merely to help when resolving many scalars from a source object - e.g. x => x.Name - which always happens synchronously. 

I know I’ve seen performance benchmarks for ObjectPool in the past but I can’t find it now.

Reply:
Actually, I talked more about memory allocations. The `ReadonlyResolveFieldContext` is needed per resolved field, right? When resolving many fields in parallel, using ObjectPool can reduce the amount of work for GC. Of course, the pool size should be limited, and maybe a custom policy is needed to sharink it after some idle time.

Reply:
Yes, per resolved field. But multiple are only needed for resolving many asynchronous fields in parallel. If the field doesn’t await a truly asynchronous method (like calling a database), fields are executed serially even with ParallelExecutionStrategy.

Reply:
There is one benefit to the current design: for serial execution strategy, the context will always be re used because it has a field per execution context rather than per thread. But with thread static it may be that the initial allocation is free.